### ISSUE TITLE
Block-scoped declarations (let, const, function, class) not yet supported outside strict mode

### ISSUE THREAD

**Issue by Thieum** [madlandproject/gulp-modernizr-build#4]
Trying to use the task in my gulp file, I get the following error when requiring the module:

```
Failed to run "C:\XXX\Gulpfile.js"...
cmd.exe /c gulp --tasks-simple
C:\XXX\node_modules\gulp-modernizr-build\src\index.js:13
let _metadataCache;
^^^
SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:404:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\XXX\gulpfile.js:16:19)
    at Module._compile (module.js:397:26)
    at Object.Module._extensions..js (module.js:404:10)
```
With the following gulpfile:

```
"use strict";
var ts = require('gulp-typescript');
var fs = require('fs');
var gulp = require('gulp');
var clean = require('gulp-clean');
var uglify = require('gulp-uglify');
var pump = require('pump');
var babel = require('gulp-babel');
var cleanCSS = require('gulp-clean-css');
const modernizr = require('gulp-modernizr-build');

var originPath = "node_modules/**";
var destPath = './libs/';
gulp.task('modernizr', function () {
    gulp.src([
        './**/*.js',
        './**/*.css',
        '!./**/modernizr*.js']) // Don't forget to exclude any other Modernizr files you may have in your sources.
        .pipe(modernizr('modernizr.js', {
            cssPrefix: 'feat-'
        }));
});

```

If I add a `"use strict";` in the index.js of your module, I get this other error:

```
Failed to run "C:\XXX\Gulpfile.js"...
cmd.exe /c gulp --tasks-simple
C:\XXX\node_modules\gulp-modernizr-build\src\index.js:82
module.exports = function(fileName, buildConfig = {}) {
                                                ^
SyntaxError: Unexpected token =
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:404:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\XXX\gulpfile.js:16:19)
    at Module._compile (module.js:397:26)
    at Object.Module._extensions..js (module.js:404:10)
```

**Comment by madlandproject** [comment]
What version of Node are you running on?

**Comment by Thieum** [comment]
ah I thought I was running 8.7.0, but I'm using node through Visual Studio, and it's using its own installation of node by default, which is version 5.4.1.



Forcing Visual Studio to use the latest version ( https://ryanhayes.net/synchronize-node-js-install-version-with-visual-studio-2015/  ), the task runs to completion with just the following warning:

```
(node:20748) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'toString' of null
(node:20748) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by madlandproject** [comment]
Ok good to know :) A quick search through the source show that there is only one call to 'toString()' when accessing file contents (or it might be an implicit call which will be longer to find). Do you have any more information on the promise rejection ? Is your build failing completely ?

**Comment by madlandproject** [comment]
Possibly fixed the unhandled rejection in v0.0.6 296ecae
